### ISSUE TITLE
"Enforce SSL on Checkout" setting irrelevant when entire site uses SSL

### ISSUE THREAD

**Issue by JJJ** [easydigitaldownloads/Easy-Digital-Downloads#8535]
### Explain your enhancement

I think that the setting with the ID `enforce_ssl` should be hidden or disabled when the entire site is already protected with an SSL certificate.

### Justification or use case

It might be confusing to users to show them a "Enforce SSL on Checkout" setting. They may think that EDD is providing something extra, when it is not.

### Mock-up

<img width="989" alt="jjj-on-2021-03-31-at-12-32-32@2x" src="https://user-images.githubusercontent.com/88951/113186381-3375d800-921d-11eb-9064-140d0fd1c206.png">

### Technical Synopsis

1. `is_ssl()` can probably safely be used to make this determination
2. `edd_checkbox_description_callback()` would need to be modified to support disabling a checkbox
3. It would probably be nice to include an updated `desc` to explain why it's disabled
4. There is no precedent yet for conditionally removing/unsetting an entire setting in EDD, from either inside or outside of `edd_get_registered_settings()`. I'm undecided if hiding it would be better or worse than exposing it but disabling it.

**Comment by JJJ** [comment]
JavaScript does exist for the `symlink_file_downloads` setting specifically, which is interesting!

https://github.com/easydigitaldownloads/easy-digital-downloads/blob/c869629d383b736a5d9cde091cc87c2d0f1a887e/assets/js/admin/settings/index.js#L117-L121

<img width="991" alt="jjj-on-2021-03-31-at-12-44-56@2x" src="https://user-images.githubusercontent.com/88951/113187835-edba0f00-921e-11eb-9b42-be3d196ba84a.png">

**Comment by robincornett** [comment]
I think it's possible for a site to be set to use SSL and still need this setting, if it's not enforcing SSL completely. My use case scenario is that I started using Software Licensing with `http` in the store URL. When I attempted to update the site to force SSL, I sent anyone who had old copies of my plugins (with the http) into redirect loops and had to back off on that, so my own site would be an example of having a certificate but still needing this to be enforced (although it's been long enough now that it may no longer be the case). I think I still have the support ticket I opened on that.

**Comment by JJJ** [comment]
@robincornett that's an interesting and very likely scenario. Thank you for mentioning it!

Ideally, http to https redirection would live inside of the web server configuration, Apache/Nginx, etc... But I do know when transitioning from one to the other, it's not unusual to have some in-between stuff, not to mention that dipping into the Apache/Nginx may not even be possible in certain web hosts.

**Comment by robincornett** [comment]
Yes, mine was done on the server level, but it broke the updater. The answer I got was this:

> We've seen this before. The problem occurs because the license activation requests are sent to `http://` then redirected by your server to `https://`, but the request parameters get lost.

I was told to approach the issue in two steps, the first being to update my plugin updater, which I did, but then never got past that. The second step involved updating the wp-config file, and I did not do that. It's been a hot minute and I still have not gotten this fully set up.

**Comment by cklosowski** [comment]
From my side, I've seen a number of times that a site was configured on `https` but the server is not configured to redirect `http` traffic to `https`. Leaving it so that the site would load on `http` if the user typed in `http://example.org`, never actually forcing. Most hosts have a 'redirect to https' setting somewhere, but I don't think we can rely on it actually existing.

I would say that if we detect the `home_url` is set to `https` that we can just make the decision to force SSL on checkout and set it to checked, and make it something that the site cannot change, but I don't think we should hide it. Maybe even show a tooltip stating why the setting is not able to be changed?
